extensions/wikia/Search/classes/Services/AbstractSearchService.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Wikia\Search\Services;
+
+
+abstract class AbstractSearchService {
+
+	public function query( $phrase ) {
+		$select = $this->prepareQuery( $phrase );
+
+		$response = $this->select( $select );
+
+		return $this->consumeResponse( $response );
+	}
+
+	protected function prepareQuery( $query ) {
+		// Subclass implements
+
+		return null;
+	}
+
+	protected function select( $select ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove the unused function parameter "$select". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)